### PR TITLE
Feature/pptt 2696 merge js api

### PR DESCRIPTION
**Pull request by prk3**
# Structure
Before:
image

After:
image

# Unified JS api document

Screenshot 2021-09-10 at 13-27-52 API — Piwik PRO Marketing Suite 16 0 documentation

### CODE REVIEW

File: data_collection/web/guides.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Guides
 ======
 
-Instalation
+Installation
```
Comment (wronan) [inline data_collection/web/guides.rst:4]: Uneven underscore.

File: data_collection/web/javascript_tracking_client/api.rst [hunk 2]
```diff
@@ -0,0 +1,1800 @@
...
+    :param number searchCount: **Optional** The number of search shown
+    :param object dimensions: **Optional** :ref:`Custom dimensions<jtc-api-custom-dimensions-object>` to pass along with the site search event
+
+    Example of usage::
+
+        [command queue]
+        _paq.push(["trackSiteSearch", "stove", undefined, 20]);
+        [tracker object]
+        tracker.trackSiteSearch("stove", undefined, 20);
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:264]: Why some many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 3]
```diff
@@ -0,0 +1,1800 @@
...
+    * track content impressions
+    * manage custom variables to use them later
+    * track clicked links to external domains and download files
+
+.. contents:: Table of Contents
+
+Installing Tracking code
+------------------------
+
+THIS SECTION WILL BE MOVED TO INSTALLATION DOCUMENT!!!!!!!!!!!!
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:24]: Should this line should indeed be published? :)

File: data_collection/web/javascript_tracking_client/api.rst [hunk 4]
```diff
@@ -0,0 +1,1800 @@
...
+
+.. _jtc-api-tracking-functions:
+
+Tracking functions
+------------------
+
+Tracking functions collect and send data to tracking backend. They can by
+called on a :ref:`tracker object<jtc-api-tracker-object>` or pushed to
+the :ref:`command queue<jtc-api-command-queue>` as commands.
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:147]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 5]
```diff
@@ -0,0 +1,1800 @@
...
+        To overwrite page title for **all events** that will happen on the page
+        (until a reload), use :ref:`setDocumentTitle<jtc-api-setDocumentTitle>`
+        function.
+
+    .. note::
+
+        ``trackPageView`` is included in the default tracker setup snippet.
+        It's likely you're already using it.
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:182]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 6]
```diff
@@ -0,0 +1,1800 @@
...
+    :param object dimensions: **Optional** :ref:`Custom dimensions<jtc-api-custom-dimensions-object>` to pass along with the custom event
+
+    Example of usage (tracking when the user clicks on the cancel button with
+    exit intent)::
+
+        [command queue]
+        _paq.push(["trackEvent", "Exit intent", "Click on button", "Cancel"]);
+        [tracker object]
+        tracker.trackEvent("Exit intent", "Click on button", "Cancel");
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:211]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 7]
```diff
@@ -0,0 +1,1800 @@
...
+    :param number conversionValue: **Optional** Conversion value (revenue)
+    :param object dimensions: **Optional** :ref:`Custom dimensions<jtc-api-custom-dimensions-object>` to pass along with the conversion
+
+    Example of usage (tracking conversion of goal *1* with value *15*)::
+
+        [command queue]
+        _paq.push(["trackGoal" 1, 15]);
+        [tracker object]
+        tracker.trackGoal(1, 15);
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:237]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 8]
```diff
@@ -0,0 +1,1800 @@
...
+    Keys in a custom dimension object must be in ``"dimensionX"`` format, where
+    ``X`` is the ID of a custom dimension. Keys that don't match this format
+    will be ignored.
+
+.. warning::
+
+    Custom dimension values **must be percent-encoded**. To encode a string,
+    pass it through ``encodeURIComponent`` function, e.g. ``encodeURIComponent("Äpfel?")``.
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:738]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 9]
```diff
@@ -0,0 +1,1800 @@
...
+        [command queue]
+        _paq.push(["trackContentImpression", "clicked", "trackingWhitepaper", "document", "http://cooltracker.tr/whitepaper"]);
+        [tracker object]
+        tracker.trackContentImpression("clicked", "trackingWhitepaper", "document", "http://cooltracker.tr/whitepaper");
+
+    .. warning::
+        Use this function in conjunction with ``trackContentImpression``, as it
+        can only be mapped with an impression by ``contentName``.
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:903]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 10]
```diff
@@ -0,0 +1,1800 @@
...
+        _paq.push(["customCrossDomainLinkVisitorIdGetter", function (url, name) {
+            return (new URL(url)).searchParams.get(name) || "";
+        }]);
+        [tracker object]
+        tracker.customCrossDomainLinkVisitorIdGetter(function (url, name) {
+            return (new URL(url)).searchParams.get(name) || "";
+        });
+
+    .. todo:: Is anyone actually overwriting the default visitor ID getter?
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:1496]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst [hunk 11]
```diff
@@ -0,0 +1,1800 @@
...
+    :param boolean enable: **Required** Whether to enable site inspector support.
+
+    Example of usage::
+
+        [command queue]
+        _paq.push(["setSiteInspectorSetup", false]);
+        [tracker object]
+        tracker.setSiteInspectorSetup(false);
+
```
Comment (wronan) [inline data_collection/web/javascript_tracking_client/api.rst:1686]: Why so many new lines?

File: data_collection/web/javascript_tracking_client/api.rst (hunk 3 above)
Comment (prk3) [reply data_collection/web/javascript_tracking_client/api.rst:24]: I will remove it.

File: data_collection/web/javascript_tracking_client/api.rst (hunk 5 above)
Comment (prk3) [reply data_collection/web/javascript_tracking_client/api.rst:182]: It's really hard to navigate a big rst document. I've added 5 newlines between sections to make it clear where a new sections starts as you scroll the file.

**Pull request merged**